src/main/java/net/dv8tion/jda/api/interactions/components/selections/SelectOption.java

```diff
@@ -208,7 +210,12 @@ public SelectOption withDefault(boolean isDefault)
     @CheckReturnValue
     public SelectOption withEmoji(@Nullable Emoji emoji)
     {
-        return new SelectOption(label, value, description, isDefault, emoji);
+        EmojiUnion union = null;
+        if (emoji instanceof EmojiUnion)
+            union = (EmojiUnion) emoji;
+        else if (emoji != null)
+            union = (EmojiUnion) Emoji.fromCustom((CustomEmoji) emoji);
```

Comment:
This is very confusing and I don't understand what is happening.
EmojiUnion is an extension of Emoji that every EmojiImpl implements. We should be able to blindly cast any `Emoji` to `EmojiUnion`, right? What is it doing with `fromCustom` here?

Reply:
Because Emoji is now an interface, users can implement their own types, which don't necessarily have the union implemented.

Reply:
We do not support external implementation of JDA interfaces.
Interfaces of entities are to hide internal implementation.